src/ContainerAppsResolver.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { ContainerAppsAPIClient, ManagedEnvironment } from "@azure/arm-appcontainers";
+import { callWithTelemetryAndErrorHandling, IActionContext, ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { AppResource, AppResourceResolver, ResolvedAppResourceBase } from "@microsoft/vscode-azext-utils/hostapi";
+import { managedEnvironmentProvider } from "./constants";
+import { ResolvedContainerAppsResource } from "./tree/ResolvedContainerAppsResource";
+import { createContainerAppsAPIClient } from "./utils/azureClients";
+import { getResourceGroupFromId } from "./utils/azureUtils";
+
+export class ContainerAppsResolver implements AppResourceResolver {
+    public async resolveResource(subContext: ISubscriptionContext, resource: AppResource): Promise<ResolvedAppResourceBase | null> {
+        return await callWithTelemetryAndErrorHandling('resolveResource', async (context: IActionContext) => {
+            const client: ContainerAppsAPIClient = await createContainerAppsAPIClient([context, subContext]);
+            const rgName: string = getResourceGroupFromId(resource.id);
```

Comment:
There should be a util function for this now in `vscode-azext-utils` that you can import (and remove utils/azureUtils) from here.

Reply:
Okay, I'm planning to change the other instances of this function call as well to sync up with this